src/com/lftechnology/java/training/alina/jdbc/dao/employee/impl/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,162 @@
+        Database database = UserService.setDatabaseParams(params);
+        String sql =
+                "Select * from user as u inner join employee as e on (u.user_id=e.user_id) where (u.user_id=? and u.is_terminated=? and e.is_deleted=?) ";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement = DbFacade.setParameterizedObjects(database, preparedStatement);
+            ResultSet result = preparedStatement.executeQuery();
+            while (result.next()) {
+                UserService userService = new UserService();
+                employeeList.add(userService.map(result));
+            }
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return employeeList;
+    }
+
+    @Override
+    public List<Employee> findAll() {
+        List<Employee> employeeList = new ArrayList<Employee>();
+        String sql =
+                "Select * from user as u inner join employee as e on (u.user_id=e.user_id) where (u.is_terminated=? and e.is_deleted=?)";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setBoolean(1, Constants.NOT_TERMINATED);
+            preparedStatement.setBoolean(2, Constants.NOT_DELETED);
+            ResultSet result = preparedStatement.executeQuery();
+            while (result.next()) {
+                UserService userService = new UserService();
+                employeeList.add(userService.map(result));
+            }
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return employeeList;
+    }
+
+    @Override
+    public Employee addNew(Employee employee) {
+        String sql = "Insert into employee (fullname,department,address,role,user_id,created_at) values (?,?,?,?,?,?)";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, employee.getFullname());
+            preparedStatement.setString(2, employee.getDepartment());
+            preparedStatement.setString(3, employee.getAddress());
+            preparedStatement.setString(4, employee.getRole());
+            preparedStatement.setInt(5, employee.getUserId());
+            preparedStatement.setTimestamp(6, employee.getCreatedAt());
+            int result = preparedStatement.executeUpdate();
+            ResultSet rs = preparedStatement.getGeneratedKeys();
+            if (result > 0) {
+                if (rs.next()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1066)